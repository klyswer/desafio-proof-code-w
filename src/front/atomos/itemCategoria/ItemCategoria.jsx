import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const ItemCategoria = ({svg,title,active}) => {
  return (
    <div className={`item_default_categoria ${active && 'item_activo'}`}>
      <div className="icon_categoria">
        <img src={`${process.env.PUBLIC_URL}/images/${svg}`} alt={title} />
      </div>
      <div className="texto_categoria">
        {title}
      </div>
    </div>
  );
}

ItemCategoria.defaultProps = {
  active:false,
}

ItemCategoria.propTypes = {
  svg:PropTypes.string.isRequired,
  title:PropTypes.string.isRequired,
  active:PropTypes.bool
};

export default ItemCategoria;