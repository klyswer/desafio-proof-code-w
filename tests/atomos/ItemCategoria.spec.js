import React from 'react';
/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import ItemCategoria from '../../src/front/atomos/itemCategoria/ItemCategoria';


describe('<ItemCategoria />', () => {


  test('Primer SnapShot', () => {
    const wrapper = shallow( <ItemCategoria title="Perfumeria" svg="aire.svg" />);
    expect(wrapper).toMatchSnapshot();
  });


  test('Caso prueba 1: props "title", debe renderizar el "string" recibido por props.', () => {
    const TitleCategory = 'Perfumer√≠a';
    const wrapper = shallow( <ItemCategoria title={TitleCategory} svg="aire.svg" />);
    const tituloCategoryRender = wrapper.find('.texto_categoria').text();

    expect(tituloCategoryRender).toBe(TitleCategory);
  });


  test('Caso prueba 2: props "svg", recibe un "string" y renderiza un svg.', () => {
    const img = 'aire.svg';
    const wrapper = shallow( <ItemCategoria svg={img} title="Perfumeria" />);
    const imgRender = wrapper.find('img').prop('src');
    
    expect(imgRender.includes(img)).toBe(true);
  });
  
  
});
