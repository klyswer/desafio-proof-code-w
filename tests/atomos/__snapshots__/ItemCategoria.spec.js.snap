// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemCategoria /> Primer SnapShot 1`] = `
<div
  className="item_default_categoria false"
>
  <div
    className="contenedor_elm_categoria"
  >
    <div
      className="icon_categoria"
    >
      <img
        alt="Perfumeria"
        src="/images/aire.svg"
      />
    </div>
    <div
      className="texto_categoria"
    >
      Perfumeria
    </div>
  </div>
</div>
`;
